tasks/main.yml

```diff
@@ -0,0 +1,422 @@
+---
+- name: install the EPEL packages required for ODCS backend
+  yum:
+    name: "{{ packages }}"
+    state: present
+    use_backend: yum
+  vars:
+    packages:
+    - python-qpid-proton
+    - python2-koji-cli-plugins
+    - python-psycopg2
+    - python2-celery
+    - mock
+    - dnf
+    - libguestfs-tools-c
+  tags:
+  - odcs
+
+- name: install the internal packages required for ODCS backend
+  yum:
+    name: "{{ packages }}"
+    state: present
+    update_cache: yes
+    use_backend: yum
+  vars:
+    packages:
+    - odcs
+    - jigdo
+    - pungi-modular-support
+    - pungi
+    - koji
+    - libmodulemd2
+  tags:
+  - odcs
+
+- name: Make sure odcs-celery-backend isn't running.
+  service:
+    name: odcs-celery-backend
+    state: stopped
+  tags:
+    - odcs
+    - odcs/backend
+
+- name: create odcs-server group
+  group:
+    name: odcs-server
+    gid: 64321
+    state: present
+ 
+- name: create odcs-server user
+  user:
+    name: odcs-server
+    uid: 64321
+    group: odcs-server
+ 
+- name: ensure ODCS service directories have right ownership
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: odcs-server
+    group: odcs-server
+    recurse: yes
+    follow: no
+  with_items:
+  - /var/run/odcs-backend
+  - /var/log/odcs-backend
+  tags:
+    - odcs
+
+#########
+# Rabbitmq part might need to be moved to different role
+# It also needs help from from fedora-infra to setup
+# https://pagure.io/fedora-infrastructure/issue/9450
+- name: copy the fedora-messaging.toml.j2 file.
+  template:
+    src: "fedora-messaging.toml.j2"
+    dest: /etc/fedora-messaging/config.toml
+    owner: odcs-server
+    group: odcs-server
+    mode: 0640
+  tags:
+  - odcs
+  
+- name: copy the odcs rabbitmq private queue crt.
+  copy:
+    src: "{{private}}/files/rabbitmq/{{env}}/pki/issued/odcs-private-queue{{env_suffix}}.crt"
```

Comment:
Since we don't have access to the fedora private stores, I can help translate these to their locations in the CentOS pkistore